onnxruntime/core/session/inference_session.cc

```diff
@@ -2192,15 +2211,13 @@ Status InferenceSession::Run(const RunOptions& run_options,
   TraceLoggingWriteStop(ortrun_activity, "OrtRun");
 #endif
 
-  // As two inference runs (one for memory allocation and one for graph capturing)
-  // are needed before replaying the captured graph, here run the inference again
-  // to capture the graph, so that users just need one session run to capture
-  // the graph.
+  // As N+1 inference runs (N for memory allocation and 1 for graph capturing)
+  // are needed before replaying the captured graph, here run N inference runs recursively until graph captured,
```

Comment:
Don't we need to call Run() recursively 2 times for the CUDA EP  because `min_num_runs_before_cuda_graph_capture_ ` is set to 2 ?

Reply:
The following code call Run() recursively (actually launches 2 regular runs) until graph is captured.